### DISCUSSION TITLE
Migration from old(Nuget.Core) to new client sdk(NuGet.Packaging & NuGet.Protocol)

### DISCUSSION THREAD

**Discussion by NavinAT** [NuGet/Home#12956] [Ideas]
In our old application we are using the following interfaces from NuGet.Core library.

```
IPackage
IPackageLookup
IPackageRepository
IPackageAssemblyReference
IPackagePathResolver
```

In the new 'NuGet.Packaging' library we could only identify the '`IPackageMetadata` & `PackageBuilder`' classes. We need the migrating information about the above mentioned interfaces in the new client SDK (NuGet.Packaging & NuGet.Protocol).

FYI: We tried using `IPackageMetadata`, thought which is equivalent to old `IPackage` interface. In our old application, we used the following two methods in many places which belongs to `IPackage`.

```
IEnumerable<IPackageFile> GetFiles();
Stream GetStream();
```

Also struggling to find the migration options for these two methods.

It will be helpful, if someone provide the guide information (any URL or sample) about all the class/interface and methods of the new nuget SDK.

Please help us here for migrating to new client SDK.

**Comment by zivkan** [comment]
I joined NuGet a long time after NuGet.Core was deprecated, so I'm not at all familiar with those interfaces. What are you using them for?

If you want to read packages, then `NuGet.Packaging`'s `PackageArchiveReader`. It has properties & methods to get files from the package, and read the nuspec to get package metadata.

**Reply by NavinAT** [reply]
Hai zivkan,
Thanks for the response.

We have several files in our application such as
1. Database which contains Sql and XML files
2. Binary Dll files,
3. Web form .ascx files 
4. Application .config files

Finally, we will pack the above mentioned files into .nupkg by using the above mentioned interfaces.
For this work we are using the old `IPackage` interface, we couldn't find how to pack the above mentioned files in new nuget client library (SDK) by just replacing the old interfaces.

Is there any possibility to just replace the old interfaces with the new one's (which we don't know) or should we have to rewrite the code by using `PackageBuilder`?.
